pkg/kube/activity.go

```diff
@@ -571,18 +571,18 @@ func GetOrCreateStage(a *v1.PipelineActivity, stageName string) (*v1.PipelineAct
 }
 
 // GetOrCreateStepInStage gets or creates the step for the given name in the given stage
-func GetOrCreateStepInStage(stage *v1.StageActivityStep, stepName string) (*v1.CoreActivityStep, bool) {
+func GetOrCreateStepInStage(stage *v1.StageActivityStep, stepName string) (v1.CoreActivityStep, bool) {
```

Comment:
What do you try to achieve here? I usually also think returning by value is better, but in this particular case I am a bit worried. Given the function name as well as the other GetOrCreate functions in _activity.go_, I'd expect a pointer. 
Now I get a copy of the step and need to somehow remember to update it in the _StageActivityStep_. In your usecase it works afaict, because you rebuild all steps in a loop.